### PR TITLE
Create initial project structure and first admin-related endpoints

### LINKED ISSUE #1: Create initial project structure, initial database models and migration system
The main goal of this task is to create the basic structure of the project, as well as make some decisions regarding database connection and unit tests structure.

Tasks:

- [x] Create initial database structure 
- [x] Figure out a Database Engine / ORM
- [x] Define how migrations will be performed
- [x] Create initial User model / table
- [x] Create initial API endpoints for CRUDing users
- [x] Create tests structure
- [x] Implement tests for the Auth API
- [x] Create a basic README with how to run and how to test

### LINKED ISSUE #7: Add authentication to admin endpoints
Add JWT authentication to the API, and ensure the admin endpoints are protected.

Tasks:
- [x] Create the `role` concept for users
- [x] Create a middleware to check for roles for each specific endpoint
- [x] Require that admin-dashboard specific endpoints are only available for staff/admin users.

### PR DESCRIPTION
**Pull request by accordeiro**
This PR:
- Closes #1 
- Closes #7

It aims to create an initial structure for this TypeScript project, providing:
- Database connection (sqlite)
- User Login / Sign up
- Authentication
- Authorization
- API endpoints for CRUDing users
- API endpoint testing
- Documentation on how to run the project

### CODE REVIEW

File: src/controllers/AuthController.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    const userRepository = getRepository(User);
+    let user: User;
+    try {
+      user = await userRepository.findOneOrFail({ where: { username } });
+    } catch (error) {
+      res.status(401).send();
+      return;
+    }
+
+    if (!user.checkIfUnencryptedPasswordIsValid(password)) {
```
Comment (ire-and-curses) [inline src/controllers/AuthController.ts]: could consider shortening name to `user.unecryptedPasswordIsValid`. I don't know the TS convention, but this is the best practice in e.g. Perl (see p.40 here ftp://cvs.happy.kiev.ua/pub/doc/Damian%20Conway%20-%20Perl%20Best%20Practices%20-%202005.pdf)

File: src/controllers/UserController.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      res.status(404).send("User not found");
+    }
+  }
+
+  public static newUser = async (req: Request, res: Response) => {
+    const { username, password, role } = req.body;
+    const user = new User();
+    user.username = username;
+    user.password = password;
+    user.role = role;
```
Comment (ire-and-curses) [inline src/controllers/UserController.ts]: Can these assignments be done as part of the `new User()` call?

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import "reflect-metadata";
+import { createConnection } from "typeorm";
+import app from "./app";
+import settings from "./settings/settings";
+
+createConnection()
+  .then(async connection => {
+    app.listen(settings.PORT, () => {
+      console.log(
+        `🚀 Anchor server has started on http://localhost:${settings.PORT}`,
```
Comment (ire-and-curses) [inline src/index.ts]: Nice touch with the rocket :rocket:

File: src/migration/1560357776337-CreateAdminUser.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { MigrationInterface, QueryRunner, getRepository } from "typeorm";
+import { User, Role } from "../entity/User";
+import { config } from "dotenv";
+
+config();
+const adminPassword = process.env.INITIAL_ADMIN_PASSWORD || "admin";
+
+export class CreateAdminUser1560357776337 implements MigrationInterface {
```
Comment (ire-and-curses) [inline src/migration/1560357776337-CreateAdminUser.ts]: What does the strangely specific number refer to? Maybe add a comment

File: README.md [hunk 5]
```diff
@@ -1 +1,19 @@
-# stellar-anchor-server
\ No newline at end of file
+# Stellar Anchor Server – Example Implementation
+
+This project is a WIP example implementation of a Stellar anchor server.
+
+Its goal is to provide a community example implementation of SEP 6 (and related SEPs 9, 10 and 12) to make it easier for anchors to integrate with the Stellar network, and enable wallets to seamlessly integrate with said anchor.
```
[the remark this answers is not in the document]
Comment (accordeiro) [reply README.md]: Done! :)

File: README.md [hunk 6]
```diff
@@ -1 +1,19 @@
...
+
+You can check the project's roadmap [here](https://github.com/stellar/stellar-anchor-server/milestones).
+
+## Running the project locally
+
+1. Install npm
```
[the remark this answers is not in the document]
Comment (accordeiro) [reply README.md]: Done – I've added the main tools' versions before the steps :)

File: src/controllers/AuthController.ts (hunk 1 above)
Comment (accordeiro) [reply src/controllers/AuthController.ts]: Agreed – I've simplified the logic / updated the naming. Thanks!

File: src/controllers/UserController.ts (hunk 2 above)
Comment (accordeiro) [reply src/controllers/UserController.ts]: Definitely! Just pushed a commit to address this.

File: src/migration/1560357776337-CreateAdminUser.ts (hunk 4 above)
Comment (accordeiro) [reply src/migration/1560357776337-CreateAdminUser.ts]: This was automatically generated by TypeORM – I've added a comment in the file to make that clearer.

**Comment by accordeiro** [comment]
Yes, I was definitely being light on types – I've changed the `tsconfig.json` to disallow implicit `any`s and implicit return `any`s, and updated the code to reflect that. Thanks for the review!

**Pull request merged**